jbi/services/jira.py

```diff
@@ -30,12 +30,23 @@
 
 JIRA_DESCRIPTION_CHAR_LIMIT = 32767
 
+
+def fatal_code(exc):
+    """Do not retry 4XX errors, mark them as fatal."""
+    try:
+        return 400 <= exc.response.status_code < 500
```

Comment:
Do we want to be more granular than this? For example, we'd want to retry on a `429 Too Many Requests` right? That's why we're using the `backoff.expo` strategy, after all.

Reply:
> For example, we'd want to retry on a 429 Too Many Requests right? 

Well, that one precisely I don't think so, since we're already sending too many requests :/ 

According to the spec, 4XX shouldn't be retried without changing the request

> That's why we're using the backoff.expo strategy, after all

Partly. If the server returns 503, we don't want to DDOS it more and wait a bit before retrying for example

Reply:
Ahh, [this](https://github.com/litl/backoff/blob/d82b23c42d7a7e2402903e71e7a7f03014a00076/README.rst#backoffruntime) is interesting